File: 백준/Gold/1074. Z/Z.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    static int count = 0;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int r = Integer.parseInt(st.nextToken());
        int c = Integer.parseInt(st.nextToken());

        int size = (int) Math.pow(2, n);
        find(size, r, c);
        System.out.println(count);
    }

    private static void find(int size, int r, int c) {
        if (size == 1)
            return;

        int half = size / 2;
        if (r < half && c < half) {
            find(half, r, c);
        } else if (r < half && c >= half) {
            count += (size * size) / 4 * 1;
            find(half, r, c - half);
        } else if (r >= half && c < half) {
            count += (size * size) / 4 * 2;
            find(half, r - half, c);
        } else {
            count += (size * size) / 4 * 3;
            find(half, r - half, c - half);
        }
    }
}

File: 백준/Gold/1074. Z/Z.kt
import kotlin.math.pow

fun main() {
    val (N, r, c) = readln().split(" ").map { it.toInt() }
    var count = 0

    fun find(r: Int, c: Int, size: Int) {
        if (size == 1) {
            return
        }

        val half = size / 2

        if (r < half && c < half) {
            find(r, c, half)
        } else if (r < half && c >= half) {
            count += (size * size) / 4 * 1
            find(r, c - half, half)
        } else if (r >= half && c < half) {
            count += (size * size) / 4 * 2
            find(r - half, c, half)
        } else if (r >= half && c >= half) {
            count += (size * size) / 4 * 3
            find(r - half, c - half, half)
        }

    }

    find(r, c, 2.0.pow(N.toDouble()).toInt())

    print(count)
}